libs/math/functions.h

```diff
@@ -152,6 +158,42 @@ sinc (T const& x)
     return (x == T(0) ? T(1) : std::sin(x) / x);
 }
 
+/* ----------------------------- Popcount ------------------------- */
+
+/** Returns the number of one bits of an integer. */
+template <typename T>
+std::size_t constexpr
+popcount (T const x)
+{
+    static_assert(std::is_integral<T>::value && std::is_unsigned<T>::value,
+        "T must be an unsigned integral type.");
+    return std::bitset<sizeof(T) * 8>(x).count();
+}
+
```

Comment:
The following conditional compilation instructions are based on the [openMVG hamming distance implementation](https://github.com/openMVG/openMVG/blob/68def23c559248bbfb64616526ebbfec341c8c1a/src/openMVG/matching/metric_hamming.hpp#L61).

Reply:
Did you got an error or a warning?
I don't have any trouble on VS2015.

I got complains about `std::uint32_t const low_bits ` if I let the `constexpr` since you can't create variable.

Reply:
I got an error just for calling _mm_popcount in a constexpr.
Not sure what causes the difference in behavior.

[Test case](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(j:1,source:'%23include+%3Ccstddef%3E%0A%23include+%3Ccinttypes%3E%0A%23include+%3Cintrin.h%3E%0A%0Astd::size_t+constexpr%0Apopcount+(std::uint32_t+const+x)%0A%7B%0A++++return+static_cast%3Cstd::size_t%3E(_mm_popcnt_u32(x))%3B%0A%7D%0A%0A//+need+a+non-inline+function+using+popcount%0A//+to+get+the+assembly%0Astd::size_t+f(std::uint32_t+x)+%7B+return+popcount(x)%3B+%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:38.56407698973815,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:cl19_32,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',trim:'0'),libs:!(),options:'',source:1),l:'5',n:'0',o:'x86+CL+19+2017+RTW+(Editor+%231,+Compiler+%231)',t:'0')),k:24.634306689968408,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:1,editor:1),l:'5',n:'0',o:'%231+with+x86+CL+19+2017+RTW',t:'0')),k:36.801616320293434,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)